beluga_demo_nav2/launch/beluga_nav2_launch.py

```diff
@@ -0,0 +1,181 @@
+
+    param_substitutions = {
+        'use_sim_time': 'True',
+        'yaml_filename': os.path.join(
+            nav2_bringup_dir, 'maps', 'turtlebot3_world.yaml'
+        ),  # noqa: E501
+    }
+
+    configured_params = ParameterFile(
+        RewrittenYaml(
+            source_file=params_file,
+            param_rewrites=param_substitutions,
+            convert_types=True,
+        ),
+        allow_substs=True,
+    )
+
+    # Declare the launch arguments
+    declare_params_file_cmd = DeclareLaunchArgument(
+        'params_file',
+        default_value=os.path.join(
+            bringup_dir, 'params', 'beluga_nav2_params.yaml'
+        ),  # noqa: E501
+        description='Full path to the ROS2 parameters'
+        'file to use for all launched nodes',
+    )
+
+    # Robot state publisher
+    urdf = os.path.join(nav2_bringup_dir, 'urdf', 'turtlebot3_waffle.urdf')
+    with open(urdf, 'r') as infp:
+        robot_description = infp.read()
+
+    start_robot_state_publisher_cmd = Node(
+        package='robot_state_publisher',
+        executable='robot_state_publisher',
+        name='robot_state_publisher',
+        output='screen',
+        parameters=[
+            {'use_sim_time': True, 'robot_description': robot_description}
+        ],  # noqa: E502
+    )
+
+    # Container for the nav2 nodes and beluga_amcl
+    start_nav2_container_cmd = Node(
+        name='nav2_container',
+        package='rclcpp_components',
+        executable='component_container_isolated',
+        parameters=[configured_params, {'autostart': True}],
+        arguments=['--ros-args', '--log-level', 'info'],
+        output='screen',
+    )
+
+    # Localization nodes
+    localization_lifecycle_nodes = ['map_server', 'amcl']
+
+    load_composable_localization_nodes = LoadComposableNodes(
+        target_container='nav2_container',
+        composable_node_descriptions=[
+            ComposableNode(
+                package='nav2_map_server',
+                plugin='nav2_map_server::MapServer',
+                name='map_server',
+                parameters=[configured_params],
+            ),
+            ComposableNode(
+                package='beluga_amcl',
+                plugin='beluga_amcl::AmclNode',
+                name='amcl',
+                parameters=[configured_params],
+            ),
+            ComposableNode(
+                package='nav2_lifecycle_manager',
+                plugin='nav2_lifecycle_manager::LifecycleManager',
+                name='lifecycle_manager_localization',
+                parameters=[
+                    {
+                        'use_sim_time': True,
+                        'autostart': True,
+                        'node_names': localization_lifecycle_nodes,
+                    }
+                ],
+            ),
+        ],
+    )
+
+    # Navigation nodes
+    navigation_lifecycle_nodes = [
+        'controller_server',
+        'planner_server',
+        'behavior_server',
```

Comment:
@Alondruck nit: considering dropping the `behavior_server`

Reply:
I thought about it, but actually, the different behavior trees implemented in Nav2 are using those action servers, and sometimes the robot might need to use them. I don't think running the behavior_server is a problem.

Reply:
Fair enough.